test/controllers/propertyLinking/ClaimPropertyOccupancyControllerSpec.scala

```diff
@@ -117,13 +117,57 @@ class ClaimPropertyOccupancyControllerSpec extends VoaPropertyLinkingSpec with S
         val doc = Jsoup.parse(contentAsString(result))
         val error = doc.getElementsByAttributeValue("href", "#lastOccupiedDate-day")
         error.text() shouldBe
-          s"Date of ${if (isAgent) "your client's " else ""}last day as " +
+          s"Date of ${if (isAgent) "your client's " else "your "}last day as " +
             s"${if (cap == Owner) "owner" else if (cap == Occupier) "occupier" else "owner and occupier"} " +
             "of the property must be after " +
             s"${d._1.getDayOfMonth} ${d._1.getMonth.toString.toLowerCase.capitalize} ${d._1.getYear}"
       }
     }
   }
+  "The claim occupancy page" should "validate that the end date of ownership is not in the future" in {
+    val dateGen: Gen[LocalDate] = Gen.choose(LocalDate.now.plusDays(1), LocalDate.now.plusYears(10))
+
+    forAll(dateGen, arbitrary[Boolean], arbitrary[CapacityType]) { (endDate, isAgent, cap) =>
+      {
+        val result = testClaimPropertyOccupancyController(
+          propertyOwnership = PropertyOwnership(fromDate = LocalDate.now),
+          relationshipCapacity = cap,
+          userIsAgent = isAgent
+        ).submitOccupancy()(FakeRequest().withFormUrlEncodedBody(
+          "stillOccupied"          -> "false",
+          "lastOccupiedDate.day"   -> endDate.getDayOfMonth.toString,
+          "lastOccupiedDate.month" -> endDate.getMonthValue.toString,
+          "lastOccupiedDate.year"  -> endDate.getYear.toString
+        ))
+        status(result) shouldBe BAD_REQUEST
+        val doc = Jsoup.parse(contentAsString(result))
+        val error = doc.getElementsByAttributeValue("href", "#lastOccupiedDate-day")
+        error.text() shouldBe
+          s"Date ${if (isAgent) "your client " else "you "}became " +
```

Comment:
i know the forAll tests all the combinations of the scenarios, but for me id rather just have different tests. Makes it easier to work out whats being tested when and what the messages are